Sources/AblyChat/PaginatedResult.swift

```diff
@@ -0,0 +1,11 @@
+public protocol PaginatedResult<T>: AnyObject, Sendable {
+    associatedtype T
+
+    var items: [T] { get }
+    var hasNext: Bool { get }
+    var isLast: Bool { get }
+    // TODO: (https://github.com/ably-labs/ably-chat-swift/issues/11): consider how to avoid the need for an unwrap
```

Comment:
I don't think this is the worst outcome for this tbh... 

we can just use `if let` which unwraps and lets you use the value within the conditional check in the same line e.g. 

```
if let nextResult = next {
    // use result of next
}
```

is there an alternative that wouldn't require checking if there is a next or not?

Reply:
> is there an alternative that wouldn't require checking if there is a next or not?

The linked #11 is about the `hasNext` property, and how it would be nice if, if `hasNext` told you "yes, there's a next page, go ahead and try to load it", then you wouldn't need to check that `next` returns a non-nil value.

Reply:
i'm indifferent, they're both very similar to me - `hasNext` still requires a bool check before being able to do anything. It's just in a property that might be more visible at a high level

Reply:
Not sure I understand your answer, but also I don't think we have to get to the bottom of this in this PR. Let's just leave #11 open and come back to this later, in the interests of getting this one merged?